File: examples/wolf_sheep/Cpp/batch_run.cpp
#include "models.h"
// #include <cppyabm/batch_runner_serial.h>
#include <cppyabm/batch_runner_parallel.h>
#include <iostream>
#include <fstream>
using namespace std;


int main(){
	auto model = make_shared<WolfSheep>();
	auto runner_obj = ParallelBatchRunner<shared_ptr<WolfSheep>>(model,20);
	runner_obj.run();
	return 0;
}

File: examples/wolf_sheep/Cppy/batch_run.py
import os
import sys
file_dir = os.path.abspath(os.path.dirname(__file__))
sys.path.insert(0,os.path.join(file_dir,'../../../scripts/'))

# from batch_runner_serial import SerialBatchRunner
from batch_runner_parallel import ParallelBatchRunner
from models import WolfSheep
import time
import pandas as pd
import time
class SETTINGS:
	iter_n = 20


if __name__ == '__main__':
	begin = time.time()
	model = WolfSheep()
	# runner_obj = SerialBatchRunner(model, iter_n=SETTINGS.iter_n)
	runner_obj = ParallelBatchRunner(model, iter_n=SETTINGS.iter_n)
	runner_obj.run()
	end = time.time()
	print("The simulation took {} seconds".format(end-begin))
